File: 02-07-2024_Linked _list_of_strings_forms_a_palindrome.py
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None

class Solution:
    def _is_palindrome(self, s):
        n = len(s)
        for i in range(n // 2):
            if s[i] != s[n - i - 1]:
                return False
        return True

    def _convert_to_string(self, head):
        temp = head
        s = ""
        while temp is not None:
            s += temp.data
            temp = temp.next
        return self._is_palindrome(s)

    def compute(self, head):
        return self._convert_to_string(head)

File: 02-07-2024_Linked _list_of_strings_forms_a_palindrome.cpp
class Solution {
  private:
    bool ispalindrome(string str)
    {
        int n=str.length();
        for(int i=0;i<n/2;i++)
        {
            if(str[i]!=str[n-i-1]) return false;
        }
        return true;
    }
    
    bool converttostring(Node* head)
    {
       Node* temp=head;
       string str;
       while(temp!=NULL)
       {
           str+=temp->data;
           temp=temp->next;
       }
       return ispalindrome(str);
    }
    
  public:
    bool compute(Node* head) {
        return converttostring(head);
    }
};
